### ISSUE TITLE
Support building on Cygwin

### ISSUE THREAD

**Issue by benalt613** [att/ast#127]
I have downloaded the package and from ast-master did a "./bin/package make" but it completes with an error:
```
/usr/bin/ksh: line 4: nmake: not found
mamake [cmd/nmake]: *** exit code 127 making Makerules.mo
mamake: *** exit code 1 making cmd/nmake
package: make: errors making /home/baltman/ast-master/arch/cygwin.i386/bin/nmake
package: make done  at Mon Nov 20 10:37:09 EST 2017 in /home/baltman/ast-master/arch/cygwin.i386
```
Has anyone been successful to build the package on cygwin? What needs to be done to get around these errors?  I have attached the output from doing the build.
[build_ksh.txt](https://github.com/att/ast/files/1488352/build_ksh.txt)

Thanks,
Ben

**Comment by siteshwar** [comment]
@benalt613 Try to compile it from ```beta``` branch.

**Comment by benalt613** [comment]
I just tried with the beta branch with similar results:
```
+ nmake --base --compile '--file=/home/baltman/ast-beta/src/cmd/nmake/Makerules.mk'
/usr/bin/ksh: line 4: nmake: not found
mamake [cmd/nmake]: *** exit code 127 making Makerules.mo
mamake: *** exit code 1 making cmd/nmake
package: make: errors making /home/baltman/ast-beta/arch/cygwin.i386/bin/nmake
package: make done  at Mon Nov 20 13:12:48 EST 2017 in /home/baltman/ast-beta/arch/cygwin.i386

```
Ben

**Comment by siteshwar** [comment]
You should paste the full build log.

**Comment by benalt613** [comment]
The output of the build is attached. 
[build_ksh.txt](https://github.com/att/ast/files/1488727/build_ksh.txt)

**Comment by siteshwar** [comment]
```
/home/baltman/ast-beta/src/lib/libast/tm/tmlocale.c:35:22: fatal error: nl_types.h: No such file or directory
```
Your system does not have ```nl_types.h``` header.

**Comment by benalt613** [comment]
Thanks for that. I found out that cygwin doesn't come with it but I did install it by including catgets package with it. After going through the build again I still get an error. I have attached the output again.
[make.ksh.2.txt](https://github.com/att/ast/files/1489097/make.ksh.2.txt)

**Comment by siteshwar** [comment]
```
/home/baltman/ast-beta/src/lib/libast/misc/spawnvex.c:802:2: error: unknown type name 'posix_spawnattr_t'
  posix_spawnattr_t  ax;
  ^~~~~~~~~~~~~~~~~
/home/baltman/ast-beta/src/lib/libast/misc/spawnvex.c:803:2: error: unknown type name 'posix_spawn_file_actions_t'
  posix_spawn_file_actions_t fx;
  ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/baltman/ast-beta/src/lib/libast/misc/spawnvex.c:808:9: error: redeclaration of 'arg' with no linkage
  int    arg;
         ^~~
/home/baltman/ast-beta/src/lib/libast/misc/spawnvex.c:799:9: note: previous declaration of 'arg' was here
  int    arg;
         ^~~
/home/baltman/ast-beta/src/lib/libast/misc/spawnvex.c:814:16: error: redeclaration of 'xev' with no linkage
  Spawnvex_t*   xev;
                ^~~
/home/baltman/ast-beta/src/lib/libast/misc/spawnvex.c:804:16: note: previous definition of 'xev' was here
  Spawnvex_t*   xev = 0;
```

You should look into all the ```errors:``` string from the output. I would suggest you to look into how/where ```'posix_spawnattr_t``` is defined under Cygwin.

**Comment by krader1961** [comment]
I see the same failure using current Cygwin with gcc on Windows 10. The */usr/include/spawn.h* header does define the `posix_spawnattr_t` type. The *./src/lib/libast/features/FEATURE/spawn* file produced by feature testing is identical to that on macOS, FreeBSD and Ubuntu. But the inclusion of *spawn.h* is predicated on several other symbols. Presumably one of those different on Cygwin.

@benalt613, If you figure out what is going wrong we'll be happy to merge a fix. Until we've finished the switch from Nmake to Meson no one else is likely to be motivated to investigate further.

**Comment by krader1961** [comment]
I just realized that issue #300 that I opened was essentially a duplicate of this issue. I've fixed the problems that make building ksh on Cygwin fail. However, there are still a huge number of unit test failures and `meson test` itself fails randomly. But since we can now build ksh on Cygwin I'm going to consider this resolved. TBD is fixing the test failures and making `meson test` more deterministic.
